pkg/plugins/secrets/k8s/store.go

```diff
@@ -50,10 +55,14 @@ func (s *KubernetesStore) Create(ctx context.Context, r *secret_model.SecretReso
 	}
 	secret.Namespace = s.namespace
 	secret.Name = opts.Name
+	labels := map[string]string{
+		meshLabel: opts.Mesh,
+	}
+	secret.SetLabels(labels)
 
 	if err := s.writer.Create(ctx, secret); err != nil {
```

Comment:
I saw that somewhere we call `SetControllerReference` for GC purposes. Why don't we do the same for secret and mesh?

Reply:
All Mesh Secrets will be removed when given Mesh is removed.

But I was also thinking about this and if we want to do it (just to be more K8S native), it should work on every entity (ie. TrafficPermission should have reference). I think first we need to make sure that https://github.com/Kong/kuma/issues/683 is done.